### DISCUSSION TITLE
Related Model Update

### DISCUSSION THREAD

**Discussion by alimo2** [phalcon/cphalcon#16336] [Q&A]
Hi

I have a table A with many secondary tables (B,C,...) related to A with one to many relation.
When I update A, one of related tables e.g C update too, but non of other updates, while relations are the same!

How can i disable it? I use `foreignKey` option, but won't work well!

My phalcon is 5.2.1 and php is 8.2.5

**Comment by alimo2** [comment]
any suggestion?

**Reply by borisdelev** [reply]
Did you invoke relation before do ```->save```?

**Reply by borisdelev** [reply]
For example

```php
        $contractor = \Models\Contractors::findFirst();
        $company = $contractor->getCompany(); // LIKE THIS?
        $contractor->save();

**Reply by alimo2** [reply]
Yes exactly. Finally i forced to findFirst() again and save() without that.

**Comment by borisdelev** [comment]
I have interesting situation.... @niden can u check out that? A help!

```php
        $contractor = \Models\Contractors::findFirst();
        $company = $contractor->getCompany();

        $contractor->save();
```

In that case, save will be executed in relation too. I mean object $contractor will be save and object $company too. Why is that?

**Reply by niden** [reply]
I will need to check that. I have said before that our ORM is nice but there are some areas where you are scratching your head on why things were coded that way.

I am hoping to be able to convert everything to PHP (the MVC namespace) so that I can thoroughly test the code and find issues like this. Doing so in Zephir is very time consuming and sometimes you cannot find what you are looking for.

I am going to add this as an issue so that I can investigate.

**Reply by borisdelev** [reply]
Thank you! I can help in PHP modules, but i need a little guide and assistance about what must be done...
